Code/6-wire-version/src/config.h

```diff
@@ -22,19 +22,27 @@ const char *OTAName = LEGACY_NAME;
 const char *OTAPassword = "esp8266";
 
 /*
- * Access Point Configuration
+ * WiFi Configuration Manager
```

Comment:
The wifi manager is good for people that don't want to edit code. And some other cases, like building a device for a friend. I've had some issues with it but if possible I want to keep it if it doesn't screw up other things. Which it does sometimes. But so does the wifi. Maybe due to my lack of understanding. Did you read the discussion about the "run without wifi"? I think it's important to have it running without wifi, and even start without wifi and reconnect when possible. But it seems you only have one shot. WiFi or autoportal. Then things go south

Reply:
you are right, the wifi manager is pretty useful. just saying that you don't the AP feature when having users who are able to hardcode the config like @hubaksis who has issues with reconnecting. there is no need to start an AP when the wifi was not found. he would have to disable the AP, otherwise the fallback is running, which is is wifi config manager AP (in this code).
guess i have a bug you stated out. that "one shot" should be fixed somewhere.
if i get it right, the loop should try to reconnect every 60 seconds using the wifi config (from wifi.json) if there is no wifi connection. cannot find anything in your code which does the reconnect. WiFi.status() does not, right?

Reply:
Yes I agree. It's a harder problem than you'd think first. Paradoxical perhaps.
List of scenarios:
No WiFi connection.
  reason a) No credentials exists.
  reason b) Credentials are wrong (maybe because you bought a new router)
  reason c) WiFi is down (router side)
How do we tell:
a) can be spotted by empty strings in WiFi.SSID() and WiFi.psk(). At least I think so.
b) can be spotted by {WiFi.status() != WL_CONNECT and the strings in a) are not empty}   ???
c) can be spotted by a scan? WiFi.SSID not in the list of networks?

What do we do:
a and b) start AP (wifi manager). Allowed to block the program until creds are entered.
c) At startup- try to connect, but do not block the code. Check periodically if WiFi is lost and in that case try to reconnect without blocking.

The whole wifi-thing in the ESP is weird to me. The OS itself tries to reconnect after a startup if you have wifi.persistant(true) and .autoreconnect(true). Or maybe it's the wifi.mode() that starts it. And then when you do several wifi.begin()/wifi.begin(ssid, pwd) things blow up sometimes. I struggled a whole day trying to sort it out but no win.

Reply:
@visualapproach please "recolve conversation". i have moved this topic to the main.cpp within this PR